src/coreclr/jit/rationalize.cpp

```diff
@@ -163,17 +163,22 @@ void Rationalizer::RewriteSIMDIndir(LIR::Use& use)
         addr->gtType = simdType;
         use.ReplaceWith(addr);
     }
-    else if (addr->OperIs(GT_ADDR) && addr->AsUnOp()->gtGetOp1()->OperIsSimdOrHWintrinsic())
+    else if (addr->OperIs(GT_ADDR))
     {
-        // If we have IND(ADDR(SIMD)) then we can keep only the SIMD node.
-        // This is a special tree created by impNormStructVal to preserve the class layout
-        // needed by call morphing on an OBJ node. This information is no longer needed at
-        // this point (and the address of a SIMD node can't be obtained anyway).
+        GenTree* nodeToCheck = addr->AsUnOp()->gtGetOp1();
 
-        BlockRange().Remove(indir);
-        BlockRange().Remove(addr);
+        if (nodeToCheck->OperIsSimdOrHWintrinsic() || nodeToCheck->IsCnsVec())
```

Comment:
Same question as in morph: given `impNormStructVal` does not create `OBJ(ADDR(CNS_VEC))`, why do we need to strip it here?